contracts/TokenSaleForAlreadyDeployedERC20Tokens.sol

```diff
@@ -0,0 +1,235 @@
+pragma solidity 0.4.24;
+
+import "./lib/Pausable.sol";
+import "./lib/FinalizableCrowdsale.sol";
+import "./lib/ERC20.sol";
+import "./Whitelist.sol";
+import "./TokenSaleInterface.sol";
+
+/**
+ * @title Token Sale contract - crowdsale of company tokens.
+ * @author Gustavo Guimaraes - <gustavo@starbase.co>
+ */
+contract TokenSaleForAlreadyDeployedERC20Tokens is FinalizableCrowdsale, Pausable {
+    uint256 public crowdsaleCap;
+    // amount of raised money in STAR
+    uint256 public starRaised;
+    uint256 public starRate;
+    bool public isWeiAccepted;
+
+    // external contracts
+    Whitelist public whitelist;
+    ERC20 public starToken;
+    // The token being sold
+    ERC20 public tokenOnSale;
+
+    event TokenRateChanged(uint256 previousRate, uint256 newRate);
+    event TokenStarRateChanged(uint256 previousStarRate, uint256 newStarRate);
+    event TokenPurchaseWithStar(address indexed purchaser, address indexed beneficiary, uint256 value, uint256 amount);
+
+    /**
+     * @dev initialization function
+     * @param _startTime The timestamp of the beginning of the crowdsale
+     * @param _endTime Timestamp when the crowdsale will finish
+     * @param _whitelist contract containing the whitelisted addresses
+     * @param _starToken STAR token contract address
+     * @param _tokenOnSale ERC20 token for sale
+     * @param _rate The token rate per ETH
+     * @param _starRate The token rate per STAR
+     * @param _wallet Multisig wallet that will hold the crowdsale funds.
+     * @param _crowdsaleCap Cap for the token sale
+     * @param _isWeiAccepted Bool for acceptance of ether in token sale
+     */
+    function init(
+        uint256 _startTime,
+        uint256 _endTime,
+        address _whitelist,
+        address _starToken,
+        address _tokenOnSale,
+        uint256 _rate,
+        uint256 _starRate,
+        address _wallet,
+        uint256 _crowdsaleCap,
+        bool    _isWeiAccepted
+    )
+        external
+    {
+        require(
+            whitelist == address(0) &&
+            starToken == address(0) &&
+            rate == 0 &&
+            starRate == 0 &&
+            tokenOnSale == address(0) &&
+            crowdsaleCap == 0,
+            "Global variables should not have been set before!"
+        );
+
+        require(
+            _whitelist != address(0) &&
+            _starToken != address(0) &&
+            !(_rate == 0 && _starRate == 0) &&
+            _tokenOnSale != address(0) &&
+            _crowdsaleCap != 0,
+            "Parameter variables cannot be empty!"
+        );
+
+        initCrowdsale(_startTime, _endTime, _rate, _wallet);
+        tokenOnSale = ERC20(_tokenOnSale);
+        whitelist = Whitelist(_whitelist);
+        starToken = ERC20(_starToken);
+        starRate = _starRate;
+        isWeiAccepted = _isWeiAccepted;
+        owner = tx.origin;
+
+        crowdsaleCap = _crowdsaleCap.mul(10 ** 18);
```

Comment:
Why this fixed definition with always 18 token decimals?

Reply:
Actually we don't need it. Will remove.